tools/tpp-run/MLIRBench.cpp

```diff
@@ -267,14 +294,37 @@ Value MLIRBench::createTimerLoop(unsigned n) {
   builder.setInsertionPointToStart(loop.getBody());
 
   // Call the kernel, ignore output
-  callKernel();
+  auto *call = callKernel();
+  assert(call && "Failed to generate a kernel call");
 
   // Revert insertion point and return the accumulation ID
   builder.setInsertionPointAfter(loop);
   return acc;
 }
 
-Value MLIRBench::getTimerStats(Value acc) {
+Value MLIRBench::getTimerStats(Value accBuf) {
+  // Skip the warmup loops in stats calculation
+  auto dimIdx = builder.create<arith::ConstantIndexOp>(unkLoc, 0);
+  auto len = builder.create<memref::DimOp>(unkLoc, accBuf, dimIdx);
+
+  // The offset is slightly inaccurate as the buffer lenght already includes
+  // the warmup iters. However, the difference is negligible for the chosen
+  // max iters and only a few more deltas will be skipped in the worst case
+  // which does not impact the overall stats.
+  auto offset = getNumWarmupIters(len);
+  auto size = builder.create<arith::SubIOp>(unkLoc, len, offset);
+  auto stride = builder.create<arith::ConstantIndexOp>(unkLoc, 1);
+  auto subview = builder.create<memref::SubViewOp>(
+      unkLoc, accBuf, ValueRange{offset}, ValueRange{size}, ValueRange{stride});
+
+  // Copy the deltas into an separate smaller buffer to ensure
+  // that perf functions do not read the warmup loop deltas
+  auto accType =
```

Comment:
Why can't you just use the subview directly?

Reply:
`perf.mean` and `perf.stdev` functions accept unranked `<*xf32>` memrefs.
AFAIK, you can't have affine map for an unranked type so the base ptr offset is lost after cast and then incorrect values are used for stats computation.

Reply:
Right, it's ok to do this copy, since it's outside of the loop and it's quite small.